src/rust/engine/process_execution/docker/src/docker.rs

```diff
@@ -816,17 +819,19 @@ impl<'a> ContainerCache<'a> {
             .await
             .map_err(|e| format!("Failed to create named cache volume for {image_name}: {e}"))?;
 
+        for mount in bind_mounts.iter_mut() {
+            mount.push_str(":ro");
```

Comment:
Could you explain why we're making these read-only?

Reply:
I see them similar to immutable inputs. Ideally, the sandboxed process shouldn't be changing the outside world.

Is there something I'm missing that gives you pause to making them read-only?

Reply:
Not concretely, just this feels like it falls somewhere between "Pants stops people shooting themselves in the foot" and "Pants gets in the way of things people need to do".

Reply:
Since we're naming them "mounts" and linking to the documentation, I feel better about widening this up.